### PR TITLE
Bugfix/376 inefficient use of entity framework

### PR DESCRIPTION
**Pull request by RamonPeek**
## Description

The current way of fetching projects executes multiple SQL queries (in a foreach). By using including the User and Likes fields in the IQueryable we only need to execute one query. This reduces the response times a lot.
Tim and I have tested this both on our local machines, and we got similar results:
- Old situation an average of 70ms.
- New situation an average of 15 to 20ms.

The Redacted User unfortunately cannot be directly included in the IQueryable, as LINQ does not support lambda statements in the Include-clause. That's why it is applied after fetching the results from the database.

## Type of change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [N.A.] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Run the API from the develop branch-source.
2. Access the endpoint /api/project via Postman.
3. Note down the average responsetime. (Be sure to access the endpoint multiple times, as the first one often takes longer to complete). I tried about 30 times.
4. Change to the current bugfix branch.
5. Re-do steps 2 and 3.

## Link to issue

Closes: #376

### CODE REVIEW

**Review by niraymak — DISMISSED** [review]
Great work! 👍

**Review by TimSnoek123 — APPROVED** [review]
Looks good to me 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Graduation system for users whose FHICT account becomes unavailable. [#287](http
 
 
 ### Fixed
-
+- Fixed inefficient use of entity framework for the fetching-all-projects endpoint.
```
Comment (Ruby77) [inline CHANGELOG.md:31]:
```suggestion
- Fixed inefficient use of entity framework for the fetching-all-projects endpoint - [376](https://github.com/DigitalExcellence/dex-backend/issues/376)
```

Small note, we add the link to the issue in the changelog.

**Review by Ruby77 — CHANGES_REQUESTED** [review]
Looks good, one small improvement. 
Congratulations on your first pull request :tada:

File: CHANGELOG.md (hunk 1 above)
Comment (RamonPeek) [reply CHANGELOG.md]: Fixed

**Review by Ruby77 — APPROVED** [review]
Well done! Your first pull request :tada: !

**Comment by Ruby77** [comment]
No problem, but I saw you forgot to add this pull request to the correct project board :wink:
image

**Review by niraymak — APPROVED** [review]
🥇

**Comment by TimSnoek123** [comment]
👍

**Review by TimSnoek123 — APPROVED** [review]
👍

**Pull request merged**